symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientBuilderWebClient.java

```diff
@@ -0,0 +1,165 @@
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withBasePath(String basePath) {
+    this.basePath = basePath;
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withUserAgent(String userAgent) {
+    withDefaultHeader("User-Agent", userAgent);
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withKeyStore(byte[] keyStoreBytes, String keyStorePassword) {
+    this.keyStoreBytes = keyStoreBytes;
+    this.keyStorePassword = keyStorePassword;
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withTrustStore(byte[] trustStoreBytes, String trustStorePassword) {
+    this.trustStoreBytes = trustStoreBytes;
+    this.trustStorePassword = trustStorePassword;
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withDefaultHeader(String key, String value) {
+    this.defaultHeaders.put(key, value);
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withTemporaryFolderPath(String temporaryFolderPath) {
+    this.temporaryFolderPath = temporaryFolderPath;
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withConnectionTimeout(int connectionTimeout) {
+    this.connectionTimeout = connectionTimeout;
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder withReadTimeout(int readTimeout) {
+    this.readTimeout = readTimeout;
+    return this;
+  }
+
+  private SslContext createSSLContext() {
+    try {
+      SslContextBuilder builder = SslContextBuilder.forClient();
+      if (this.trustStoreBytes != null) {
+        final KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustStore.load(new ByteArrayInputStream(this.trustStoreBytes), this.trustStorePassword.toCharArray());
+        trustManagerFactory.init(trustStore);
+        builder.trustManager(trustManagerFactory);
+        if (this.keyStoreBytes != null) {
+          final KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+          final KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+          keyStore.load(new ByteArrayInputStream(this.keyStoreBytes), this.keyStorePassword.toCharArray());
+          keyManagerFactory.init(keyStore, this.keyStorePassword.toCharArray());
+          builder.keyManager(keyManagerFactory);
+        }
+      }
+      return builder.build();
+    } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException | UnrecoverableKeyException e) {
+      return null;
```

Comment:
should we throw it as a runtime exception to be able to find out the root cause?